extensions/amp-a4a/0.1/amp-ad-template-helper.js

```diff
@@ -82,7 +82,7 @@ export class AmpAdTemplateHelper {
    * @return {!Promise<!Element>} Promise which resolves after rendering completes.
    */
   render(templateValues, element) {
-    return Services.templatesForDoc(this.ampdoc_).findAndRenderTemplate(
+    return Services.templatesForDoc(element).findAndRenderTemplate(
```

Comment:
Should we rename `this.ampdoc_` to make it clear that it refers to parent ampdoc?

Reply:
Alternatively, we could resolve the whole helper from the context element. Would that be better? E.g. instead of
```
getAmpAdTemplateHelper(this.iframe)
```

We'd do:
```
getAmpAdTemplateHelper(this.getDocument(this.iframe).body)
```

Reply:
An alternative diff:

```
diff --git a/extensions/amp-a4a/0.1/template-renderer.js b/extensions/amp-a4a/0.1/template-renderer.js
index aa9819de1..347d0d418 100644
--- a/extensions/amp-a4a/0.1/template-renderer.js
+++ b/extensions/amp-a4a/0.1/template-renderer.js
@@ -72,9 +72,10 @@ export class TemplateRenderer extends Renderer {
       if (!data) {
         return Promise.resolve();
       }
-      const templateHelper = getAmpAdTemplateHelper(element);
+      const iframeBody = this.getDocument(iframe).body;
+      const templateHelper = getAmpAdTemplateHelper(iframeBody);
       return templateHelper
-        .render(data, this.getDocument(iframe).body)
+        .render(data, iframeBody)
         .then((renderedElement) => {
           const {analytics} = templateData;
           if (analytics) {
```

Reply:
Renamed to `parentAmpdoc`. Gonna keep helper resolution unchanged.